### PR TITLE
feat: [redis] add patch feature and userdefined types

### PR DESCRIPTION
**Pull request by pie-r**
## Description

In this  PR I'm adding a new feature that allow to configure the redis patch allowed time and using user-deifned types to simplify the module.

For diagnosticSettings the module is less opinionated, this is the main difference as a result from the user defined types that don't allow default values.

## Updating an existing module

- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [ x I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ x] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: modules/storage/redis-cache/README.md [hunk 1]
```diff
@@ -4,43 +4,39 @@ This module deploys Azure Cache for Redis(Microsoft.Cache/redis) and optionally
 
 ## Details
 
-[Azure Cache for Redis](https://azure.microsoft.com/en-us/pricing/details/cache/)  Azure Cache for Redis gives you the ability to use a secure open source Redis cache. It's a dedicated offering managed by Microsoft, to build highly scalable and responsive applications by providing you super-fast access to your data. You get to leverage the rich feature set and ecosystem provided by Redis
+{{ Add detailed information about the module. }}
```
Comment (shenglol) [inline modules/storage/redis-cache/README.md:7]: Can you restore the details section?

Comment (shenglol) [reply modules/storage/redis-cache/README.md:7]: BTW, is the Details section removed by `brm generate`? If yes, we may have a bug.

File: modules/storage/redis-cache/README.md [hunk 2]
```diff
@@ -70,7 +66,7 @@ module redisCache 'br/public:data/redis-cache:1.0.1' = {
 ### Example 2
 
 ```bicep
-module redisCache 'br/public:data/redis-cache:1.0.1' = {
+module redisCache 'br/public:data/redis-cache:2.0.1' = {
```
Comment (shenglol) [inline modules/storage/redis-cache/README.md:69]:
```suggestion
module redisCache 'br/public:storage/redis-cache:2.0.1' = {
```

File: modules/storage/redis-cache/README.md (hunk 1 above)
Comment (pie-r) [reply modules/storage/redis-cache/README.md]: fixed, there's no bug. it has been my fault yesterday.

Comment (shenglol) [reply modules/storage/redis-cache/README.md]: Oh got it. Thanks for confirming!

**Review by shenglol — APPROVED** [review]
:shipit:

**Pull request merged**